### DISCUSSION TITLE
Side Panel - Details form block - Better Performance when Ejected

### DISCUSSION THREAD

**Discussion by mmalessa80** [Budibase/budibase#10285] [Help] [answered]
Now that someone has assisted in correctly setting up a table with a side panel I have started exploring the side panel functionality a little more. One of the curious things I came across is when leaving the 'Details form block' under the 'Details side panel' and clicking on different rows in the table in the background, there is a short lag in switching the row data on the side panel.

I then ejected the 'Details form block' and to my surprise there no longer is a lag when clicking on different rows in the table in the background.

I was hoping that someone with better base knowledge of how Budibase works might be able to explain the logic to me behind this? Primarily I was actually looking at ejecting the side panel so I can control the order of the row columns along with which ones are being displayed which apparently the order cannot be controlled unless the 'Details form block' is ejected.

**Comment by andz-bb** [comment, marked as answer]
Hey @mmalessa80 

I haven't been able to notice any difference in performance between standard form blocks and ejected ones. Whenever you click on a table row to open the side panel with the details form, Budibase sends a POST request to the search endpoint and then a GET request to the rows endpoint. The loading time for the side panel form content should be limited by the response times from those requests, and the content of the requests doesn't appear to change whether the form is inside a block or not.

In this example I click the first four rows on my table and these are the requests sent to the server:
**ejected table block with intact details form block**
![ejected table block with intact details form block](https://user-images.githubusercontent.com/110921612/231723566-39ad9933-c0c6-4056-ab11-bb3cf85f9213.png)

**ejected table block with ejected details form block**
![ejected table block with ejected details form block](https://user-images.githubusercontent.com/110921612/231723555-96e1783c-a4bb-4ae2-bc16-61be951ab067.png)

As you can see the times don't vary significantly and will depend on the server's performance at that moment rather than anything in the front end. 

All of that being said I have only spent a short time investigating this and don't know all of the ins and outs of blocks from a technical perspective. If you have a working example demonstrating the difference in performance between the two I'd love to see it so that I can investigate further. If you have an app export or a video showing a recreate for this please send it through and I'll have a deeper look :+1:

**Reply by mmalessa80** [reply]
Let me investigate this a little further. Could you tell me what tool you are using to measure the performance? That way I can see if there is a significant difference before I go further down the rabbit hole. I can definitely say that the non-ejected form block side panel even shows a spinner temporarily on initial row click. Then while the side panel remains open there is a distinguishable pause when clicking on a different table row.

**Reply by andz-bb** [reply]
thanks @mmalessa80, I'm just using the Chrome developer tools here. That section I'm showing is the Network tab where I have it filtered to only show Fetch/XHR requests. The time and waterfall indicate when each request is made and how long they take. You should be able to access this menu by pressing F12 in the browser or going to View > Developer > Developer tools in the menu.
